### ISSUE TITLE
Hyperparameter tuning for an LGBM classifier

### ISSUE THREAD

**Issue by dblim** [microsoft/LightGBM#6275]
## Description
Hi, so this isn't as much of a bug as a question I have about LGBM. I am tuning hyperparameters with 3-fold cross validation for an LGBM classifier on a dataset that has about 2 million samples with 100 features. My question is this. I can basically let n_estimators grow very very large (say about 5000) while making the learning rate get small (say about 0.001), while decreasing my cv negative log-likelihood. What's going on here? 

I have tried searching online about advice on LGBM hyperparameter tuning, but the advice seems to be generic and doesn't match what I'm looking for. I guess I'd like to know if anyone has come across this issue with LGBM before? 

## Environment info

LightGBM version or commit hash:

Command(s) you used to install LightGBM

```shell

```
